components/brave_wallet_ui/panel/async/wallet_panel_async_handler.ts

```diff
@@ -532,7 +537,7 @@ handler.on(PanelActions.signMessageHardware.getType(), async (store, messageData
   const payload: SignMessageProcessedPayload =
     signed.success
       ? { approved: signed.success, id: messageData.id, signature: signature }
-      : { approved: signed.success, id: messageData.id, error: signed.error }
+      : { approved: signed.success, id: messageData.id, error: (signed.error) as string }
```

Comment:
maybe just call .toString() instead of casting?

Reply:
Tried, but won't work here just because the signed.error is of type `error?`, and toString is not defined on undefined. `signed` is a HardwareOperationalResult:

```
export type HardwareOperationResult = {
  success: boolean
  error?: string | LedgerError
  code?: string | number
}
```

So this reduces to the same issue I raise [here](https://github.com/brave/brave-core/pull/14096/files#r923829402).  I am now wondering if I should not try to propagate the full errors (complete with statusCodes) to getErrorMessage in components/brave_wallet_ui/components/desktop/popup-modals/add-account-modal/hardware-wallet-connect/index.tsx, and instead continue just passing strings up.  That way we only have one error type (string) and can eliminate the casts.  That would be at the expense of being able to case on statusCode in getErrorMessage.  Thoughts?

Reply:
This cast should actually be `as string | undefined` to rule out LedgerError which is the goal of this cast.  `as string` is actually too narrow since it legitimately may be undefined give the above HardwareOperational result type.  I updated in https://github.com/brave/brave-core/pull/14096/commits/499d2df564ab1a0c3de85642c195a9f5b8552b15.